src/SonarScanner.MSBuild.Shim/PropertiesWriter.cs

```diff
@@ -250,11 +250,15 @@ public void WriteSonarProjectInfo(DirectoryInfo projectBaseDir)
             AppendKeyValue(SonarProperties.PullRequestCacheBasePath, config.GetConfigValue(SonarProperties.PullRequestCacheBasePath, null));
         }
 
-        public void WriteSharedFiles(IEnumerable<FileInfo> sharedFiles)
+        public void WriteSharedFiles(IEnumerable<FileInfo> sources, IEnumerable<FileInfo> tests)
         {
-            if (sharedFiles.Any())
+            if (sources.Any())
```

Comment:
We do multiple evaluations for both parameters. If at some point we will refactor the code and return an IEnumerable that is lazy evaluated (like the result of Directory.EnumerateFiles), then the evaluation will be done twice and we will not notice.

I would prefer to use `ICollection<FileInfo>` instead.